src/main/java/net/dv8tion/jda/internal/requests/Requester.java

```diff
@@ -329,4 +309,8 @@ public void shutdown()
         rateLimiter.shutdown();
     }
 
+    private static boolean shouldRetry(int code)
+    {
+        return code == 502 || code == 504 || code == 529;
```

Comment:
so why aren't we retrying 500s?

Reply:
Because it likely never succeeds when a server error happened. And usually just results in double send.